pingora-core/src/apps/mod.rs

```diff
@@ -52,6 +52,11 @@ pub trait ServerApp {
 
     /// This callback will be called once after the service stops listening to its endpoints.
     fn cleanup(&self) {}
+
+    /// This callback will be called once after the service stops listening to its endpoints.
+    ///
+    /// Used for async cleanup tasks.
+    async fn async_cleanup(&self) {}
```

Comment:
The async-drop utilizes calling block_in_place within a synchronous drop, which turns the current thread into a blocking one to execute the asynchronous task. This mechanism cannot be used in a current_thread runtime, as it would cause a [panic.](https://docs.rs/tokio/latest/tokio/task/fn.block_in_place.html#panics)